hypothesis-python/src/hypothesis/ghostwriter.py

```diff
@@ -0,0 +1,192 @@
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis/
+#
+# Most of this work is copyright (C) 2013-2020 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+"""
+WARNING: this module is under development and should not be used... yet.
+See https://github.com/HypothesisWorks/hypothesis/pull/2344 for progress.
+"""
+
+import builtins
+import enum
+import inspect
+from collections import OrderedDict
+from textwrap import indent
+from typing import Callable, Dict, Tuple, Type, Union
+
+import hypothesis.strategies as st
+from hypothesis.errors import InvalidArgument
+from hypothesis.utils.conventions import InferType, infer
+
+IMPORT_SECTION = """
+# This test template was produced by the `hypothesis.ghostwriter` module.
+{imports}
+"""
+
+TEMPLATE = """
+{nothing_hint}@given({given_args})
+def test_{test_kind}_{func_name}({arg_names}):
+{test_body}
+"""
+
+SUPPRESS_BLOCK = """\
+try:
+{test_body}
+except {exceptions}:
+    assume(False)
```

Comment:
Why are we using `assume(False)` instead of `reject()`?

Reply:
Fixed.